floyd/manager/experiment_config.py

```diff
@@ -26,5 +26,25 @@ def get_config(cls):
             raise FloydException("Missing .floydexpt file, run floyd init first")
 
         with open(cls.CONFIG_FILE_PATH, "r") as config_file:
-            experiment_config_str = config_file.read()
-        return ExperimentConfig.from_dict(json.loads(experiment_config_str))
+            experiment_config = json.loads(config_file.read())
+
+        # Get the family_id from the API and write it to the file
```

Comment:
I'm not a big fan of a get operation updating the config as a side effect. Can you explicitly check for this from the run() command function and if the family_id does not exist, then populate the id explicitly?

Reply:
Yeah you have a good point. This should be changed.

On the other side of the coin, having every part of our code that deals with the experiment config have to check if the config has an id, then get the id using the project client, set the id on the ExperimentConfig object, and write it back to disk seems not so good either. Maybe something like a "get_or_create_config" method or something. But QP brought up a good point that maybe we can avoid having to save the config without the id in the first place. Let me think on it a bit and come up with something better.